### PR TITLE
chore: remove vimscript

### LINKED ISSUE #101: Migrating to Lua
track migration to lua progress

### PR DESCRIPTION
**Pull request by dhananjaylatkar**
fixes #101 #90

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ if you're using Lua for your configs. Read the documentation below to learn
 more about specific options (*NOTE:* options in the `setup` function don't
 have the `gitblame_` prefix).
 
-> **NOTE:** you don't have to call `setup` if you don't want to customize the
+> **NOTE:** you must call `setup` if even if you don't want to customize the
```
Comment (f-person) [inline README.md:71]: Would calling `setup` on the lib side once be an issue?

File: lua/gitblame/config.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
+local M = {}
+
+M.default_opts = {
+    enabled = true,
```
Comment (dhananjaylatkar) [inline lua/gitblame/config.lua:4]: Order these same as in README

File: README.md (hunk 1 above)
Comment (dhananjaylatkar) [reply README.md:71]: yes. `enabled` option is creating issue.

Comment (f-person) [reply README.md:71]: 🤔
What if we disable in setup if `enabled` is false?

Comment (dhananjaylatkar) [reply README.md:71]: That disable call is not clearing the virtual text.

Here is what's happening -
Default is `enabled=true` when we run `setup()` in `plugin/gitblame.lua` it enabled the plugin.
When user calls `setup()` with `enabled=false`, `setup()` calls `disable()` which calls `clear_all_extmarks()`.
But `clear_all_extmarks()` is not finding all buffers for some reason and virtual text for buffer is not cleaned.

Comment (f-person) [reply README.md:71]: Hmm, this sounds like a bug :)

Comment (dhananjaylatkar) [reply README.md]: It was problem with `show_blame_info()`. 
1st `setup()` calls `show_blame_info` as callback of vim.schedule(). \
Which is asynchronous call, and takes some time to reach callback. 
So 2nd `setup()` call with `enabled=false` executes `disable()` before `show_blame_info`.
Which is why blame info was showing for one line.

It is fixed with latest commit

Comment (bossley9) [reply README.md]: Now that the issue is resolved, would we be able to disable the plugin in setup if `enabled` is false? Ideally we want to try to maintain backwards compatibility as much as possible, and requiring `setup` might cause friction for users updating the plugin :slightly_smiling_face:

Comment (dhananjaylatkar) [reply README.md]: Yep. It works!

File: lua/gitblame/init.lua [hunk 3]
```diff
@@ -563,18 +564,62 @@ local function clear_all_extmarks()
...
+
+M.disable = function(force)
+    if not vim.g.gitblame_enabled and not force then
         return
     end
 
-    vim.g.gitblame_enabled = 0
-
+    vim.g.gitblame_enabled = false
+    pcall(vim.api.nvim_del_augroup_by_name, "gitblame")
```
Comment (bossley9) [inline lua/gitblame/init.lua:644]:
```suggestion
    vim.api.nvim_del_augroup_by_name("gitblame")
```

**Review by bossley9 — COMMENTED** [review]
Looks good to me!

File: lua/gitblame/init.lua (hunk 3 above)
Comment (dhananjaylatkar) [reply lua/gitblame/init.lua:644]: `vim.api.nvim_del_augroup_by_name("gitblame")` fails when group does not exists. This happens when enabled is false. So using `pcall()`

Comment (bossley9) [reply lua/gitblame/init.lua:644]: Ah, gotcha!

**Comment by f-person** [comment]
Hooray! Thank you, @dhananjaylatkar & @bossley9! <3


https://github.com/f-person/git-blame.nvim/assets/31893391/a1429418-ab9e-4d15-b4e1-309ffa889e35

**Pull request merged**